var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -110,8 +110,20 @@ def install(self, spec, prefix):
                         '-rhsone')
             make("install")
 
+    @property
+    def headers(self):
+        """Export the main hypre header, HYPRE.h; all other headers can be found
+        in the same directory.
+        Sample usage: spec['hypre'].headers.cpp_flags
+        """
+        hdrs = HeaderList(find(self.prefix.include, 'HYPRE.h', recursive=False))
+        return hdrs or None
+
     @property
     def libs(self):
-        is_shared = self.spec.satisfies('+shared')
```

Comment:
I don’t remember details, but this and the new way are not always the same.

Reply:
Which one is the correct (or just preferred)?

Reply:
@alalazo probably knows the difference 😄

Reply:
@v-dobrev Reading this discussion right now. I think @davydden refers to #4135

Reply:
yes, thanks for the link, @alalazo . I rembembered that there was a discussion somewhere